### PR TITLE
Non-Threaded State Machine

### PR DESCRIPTION
**Pull request by wavious-wpatty**
Fixes:
  - None

Features:
  - Added a non-threaded state machine implementation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,3 +28,23 @@ can be started by running the following command from the root of this project:
...
 
 ---
+
+## FSM vs StateMachine
+FSM provides a dedicated task that all FSMs share. FSM events are submitted
+to a queue and processed directly by the FSM task, which move the FSM
+between diferrent state. This can be nice to have when the software has
+many FSMs that need to be used but doesn't need tight control of the
+run loop. It can save memory by having all events processed in a single
+place and dispatched the the approriate FSM.
```
Comment (wavious-jbasista) [inline README.md:38]: I think this is meant to say "dispatch the appropriate FSM."

File: include/kernel/stateMachine.h [hunk 2]
```diff
@@ -0,0 +1,487 @@
+/*
```
Comment (wavious-jbasista) [inline include/kernel/stateMachine.h:1]: I am assuming this file is a copy of the original, therefore we will not want to change anything in it for now.  If this is true, will not review this file and will leave as is.

File: src/kernel/stateMachine.c [hunk 3]
```diff
@@ -0,0 +1,178 @@
+/*
```
Comment (wavious-jbasista) [inline src/kernel/stateMachine.c:1]: Same as the comment in the .h file.

File: README.md (hunk 1 above)
Comment (wavious-wpatty) [reply README.md:38]: It should read "dispatched to the appropriate FSM".

File: include/kernel/stateMachine.h (hunk 2 above)
Comment (wavious-wpatty) [reply include/kernel/stateMachine.h:1]: Yes.

File: src/kernel/stateMachine.c (hunk 3 above)
Comment (wavious-wpatty) [reply src/kernel/stateMachine.c:1]: yes.

**Review by wavious-jbasista — APPROVED** [review]

**Pull request merged**